From 2d4771ba17d5547e4d3c4a07441fee3783ce5a3f Mon Sep 17 00:00:00 2001
From: Thierry FOURNIER <tfournier@exceliance.fr>
Date: Fri, 6 Dec 2013 10:39:48 +0100
Subject: [PATCH 288/311] MINOR: map: export map_get_reference() function

This function is used to identify map with his reference into the CLI
functions.
---
 include/proto/map.h | 29 +++++++++++++++++++++++++++++
 src/map.c           |  2 +-
 2 files changed, 30 insertions(+), 1 deletion(-)
 create mode 100644 include/proto/map.h

diff --git a/include/proto/map.h b/include/proto/map.h
new file mode 100644
index 0000000..4dd081b
--- /dev/null
+++ b/include/proto/map.h
@@ -0,0 +1,29 @@
+/*
+ * include/proto/map.h
+ * This file provides structures and types for pattern matching.
+ *
+ * Copyright (C) 2000-2013 Willy Tarreau - w@1wt.eu
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, version 2.1
+ * exclusively.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef _PROTO_MAP_H
+#define _PROTO_MAP_H
+
+#include <types/map.h>
+
+struct map_reference *map_get_reference(const char *reference);
+
+#endif /* _PROTO_PATTERN_H */
diff --git a/src/map.c b/src/map.c
index 78392d4..fc961e7 100644
--- a/src/map.c
+++ b/src/map.c
@@ -25,7 +25,7 @@
 struct list maps = LIST_HEAD_INIT(maps); /* list of struct map_reference */
 
 /* This function return existing map reference or return NULL. */
-static struct map_reference *map_get_reference(const char *reference)
+struct map_reference *map_get_reference(const char *reference)
 {
 	struct map_reference *ref;
 
-- 
1.8.1.5

